src/v3/bandwidthRtc.ts

```diff
@@ -0,0 +1,445 @@
+    this.cleanupPublishedStreams();
+  }
+
+  private async offerPublishSdp(): Promise<PublishSdpAnswer> {
+    const localSdpOffer = await this.publishingPeerConnection!.createOffer({
+      offerToReceiveVideo: false,
+      offerToReceiveAudio: false,
+      voiceActivityDetection: true,
+      iceRestart: false,
+    });
+
+    const publishMetadata = [...this.publishedStreams].reduce((publishMetadata: PublishMetadata, [streamId, stream]) => {
+      publishMetadata[streamId] = stream.metadata;
+      return publishMetadata;
+    }, {});
+    const remoteSdpAnswer = await this.signaling.offerSdp(localSdpOffer.sdp!, publishMetadata);
+
+    await this.publishingPeerConnection!.setLocalDescription(localSdpOffer);
+    await this.publishingPeerConnection!.setRemoteDescription({
+      type: "answer",
+      sdp: remoteSdpAnswer.sdpAnswer,
+    });
+
+    return remoteSdpAnswer;
+  }
+
+  private async handleSubscribeSdpOffer(subscribeSdpOffer: SubscribeSdpOffer) {
+    logger.info("Received SDP offer", subscribeSdpOffer);
+    if (subscribeSdpOffer.sdpRevision <= this.subscribingPeerConnectionSdpRevision) {
+      logger.debug(
+        `Revision on SDP offer (${subscribeSdpOffer.sdpRevision}) is less than current revision (${this.subscribingPeerConnectionSdpRevision}), ignoring`
+      );
+      return;
+    } else {
+      this.subscribingPeerConnectionSdpRevision = subscribeSdpOffer.sdpRevision;
+    }
+    const remoteSdpOffer = subscribeSdpOffer.sdpOffer;
+
+    this.subscribedStreams.clear();
+    Object.entries(subscribeSdpOffer.streamMetadata).forEach(([streamId, metadata]) => {
+      this.subscribedStreams.set(streamId, metadata);
+    });
+    // this.inboundStreams = new Map(Object.entries(inboundSdpOffer.streamMetadata));
+
+    if (!this.subscribingPeerConnection) {
+      await this.setupSubscribingPeerConnection();
+    }
+
+    await this.subscribingPeerConnection!.setRemoteDescription({
+      type: "offer",
+      sdp: remoteSdpOffer,
+    });
+
+    let localSdpAnswer = await this.subscribingPeerConnection!.createAnswer();
+    if (!localSdpAnswer.sdp) {
+      throw new Error(`RTCPeerConnection.createAnswer returned ${localSdpAnswer}`);
+    }
+
+    // Munge the SDP to change the setup from "actpass" to "passive"
+    // This unfortunately seems to be required
+    let parsedSdpAnswer = sdpTransform.parse(localSdpAnswer.sdp!);
+    parsedSdpAnswer.media.forEach((media) => (media.setup = "passive"));
+    localSdpAnswer.sdp = sdpTransform.write(parsedSdpAnswer);
+
+    // TODO: see if this actually needs to be awaited
+    await this.signaling.answerSdp(localSdpAnswer.sdp);
+
+    await this.subscribingPeerConnection!.setLocalDescription(localSdpAnswer);
+  }
+
+  private async setupPublishingPeerConnection(): Promise<RTCPeerConnection> {
+    this.publishingPeerConnection = new RTCPeerConnection(RTC_CONFIGURATION);
+
+    this.setupNewPeerConnection(this.publishingPeerConnection, this.setupPublishingPeerConnection);
```

Comment:
Is it intentional to pass this function (setupPublishingPeerConnection()) as the handler for onPeerClosed?

Reply:
Yep, the idea is that the peers will "respawn" if they die, and repeat all of their initial setup.